### DISCUSSION TITLE
Use Stylet in a single instance application via Mutex

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#328] [Q&A] [answered]
In a typical WPF application, the `Main()` function was automatically generated by setting build action of `App.xaml` to be "ApplicationDefinition" and Stylet works great with such settings.  However, I have a particular case that the WPF app needs to be a single-instance application, it can be achieved fairly easily with the `Mutex` library and below is the code snippet of the custom `Main()` function.  To prevent the custom `Main()` function being overwritten by the code generator, the build action of `App.xaml` needs to be changed to "Page" and Stylet stops working under such circumstance.  How should I make it work with Stylet?  Please advise, thank you very much!

```
        private static Mutex mutex = new Mutex(true, "{0c21e65e-6cc4-11ec-90d6-0242ac120003}");

        /// <summary>
        /// Application Entry Point.
        /// </summary>
        [System.STAThreadAttribute()]
        [System.Diagnostics.DebuggerNonUserCodeAttribute()]
        [System.CodeDom.Compiler.GeneratedCodeAttribute("PresentationBuildTasks", "4.0.0.0")]
        public static void Main()
        {
            if (mutex.WaitOne(System.TimeSpan.Zero, true))
            {
                SpatSuite.App app = new SpatSuite.App();
                app.InitializeComponent();
                app.Run();
                mutex.ReleaseMutex();
            }
            else
            {
                MessageBox.Show("Instance has been running!");
            }
        }
```

**Comment by canton7** [comment, marked as answer]
Why not keep the entry point as Application, but do the mutex check in the Bootstrapper's OnStart method (or perhaps later, e.g. in Configure)?

Another option is to write your own Main, then set it as the entry point in the project properties (in Application -> General -> Startup object). That means you can keep your App.xaml set as an ApplicationDefinition. 

If neither of those work, take a look at the code that WPF is generating (in obj), and find the difference between the outputs for ApplicationDefinition and Page.

**Reply by mwdpb** [reply]
Thanks for the quick reply.  Putting mutex check in Bootstrapper's `Configure` works great!
